x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -249,28 +303,238 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
+        });
+      } else {
+        // This is an edge case and should never happen. But just in case :-)
+        warnings.push(
+          `${command} response action ID [${action.EndpointActions.action_id}] missing SentinelOne agent ID, thus unable to check on it's status. Forcing it to complete as failure.`
+        );
+
+        completedResponses.push(
+          this.buildActionResponseEsDoc<{}, SentinelOneIsolationResponseMeta>({
+            actionId: action.EndpointActions.action_id,
+            agentId: Array.isArray(action.agent.id) ? action.agent.id[0] : action.agent.id,
+            data: { command },
+            error: {
+              message: `Unable to very if action completed. SentinelOne agent id ('meta.agentId') missing on action request document!`,
+            },
+          })
+        );
+      }
+
+      return acc;
+    }, [] as QueryDslQueryContainer[]);
+
+    if (agentListQuery.length > 0) {
+      const query: QueryDslQueryContainer = {
+        bool: {
+          must: [
+            {
+              terms: {
+                // Activity Types can be retrieved from S1 via API: `/web/api/v2.1/activities/types`
+                'sentinel_one.activity.type':
+                  command === 'isolate'
+                    ? [
+                        // {
+                        //    "id": 1001
+                        //    "action": "Agent Disconnected From Network",
+                        //    "descriptionTemplate": "Agent {{ computer_name }} was disconnected from network.",
+                        // },
+                        1001,
+
+                        // {
+                        //    "id": 2010
+                        //    "action": "Agent Mitigation Report Quarantine Network Failed",
+                        //    "descriptionTemplate": "Agent {{ computer_name }} was unable to disconnect from network.",
+                        // },
+                        2010,
+                      ]
+                    : [
+                        // {
+                        //    "id": 1002
+                        //    "action": "Agent Reconnected To Network",
+                        //    "descriptionTemplate": "Agent {{ computer_name }} was connected to network.",
+                        // },
+                        1002,
+                      ],
+              },
+            },
+          ],
+          should: agentListQuery,
+          minimum_should_match: 1,
+        },
+      };
+
+      const searchRequestOptions: SearchRequest = {
+        index: SENTINEL_ONE_ACTIVITY_INDEX,
+        query,
+        // There may be many documents for each host/agent, so we collapse it and only get back the
+        // first one that came in after the isolate request was sent
+        collapse: {
+          field: 'sentinel_one.activity.agent.id',
+          inner_hits: {
+            name: 'first_found',
+            size: 1,
+            sort: [{ 'sentinel_one.activity.updated_at': 'asc' }],
```

Comment:
I believe, the sort order for `sentinel_one.activity.updated_at` in the `inner_hits` as well as the outer query should be `desc` so that the most recent item is first in the array/list.

Reply:
no... it should be `asc` here based on the approach we need to take to determine if the action was complete. I can explain more if you like, but essentially: you want the **first** document that was received where the `updated_at` value is greater than when the response action was sent